module/advancement/advancement-config.mjs

```diff
@@ -76,6 +80,15 @@ export default class AdvancementConfig extends FormApplication {
 
   /* -------------------------------------------- */
 
+  activateListeners(html) {
+    super.activateListeners(html);
+
+    // Remove an item from the list
+    if ( this.options.dropKeyPath ) html.on("click", ".item-delete", this._onItemDelete.bind(this));
```

Comment:
Not sure if we envision this app might in the future have other sort of control buttons, but if so I do prefer the paradigm of handling all such buttons with one aggregated handler, usually with the class selector `.action-button` or similar and using `data-action` in the HTML markup to differentiate button intent.